src/library/tabledelegates/stardelegate.h

```diff
@@ -31,9 +33,27 @@ class StarDelegate : public TableItemDelegate {
 
   private slots:
     void commitAndCloseEditor();
+    void closingEditor(QWidget* widget, QAbstractItemDelegate::EndEditHint hint);
     void cellEntered(const QModelIndex& index);
+    void cursorNotOverAnyCell();
+    void editRequested(const QModelIndex& index,
+            QAbstractItemView::EditTrigger trigger,
+            QEvent* event);
+    void restorePersistentRatingEditorNow();
 
   private:
+    void openPersistentRatingEditor(const QModelIndex& index);
+    void closeCurrentPersistentRatingEditor(bool rememberForRestore);
+    void restorePersistentRatingEditor(const QModelIndex& index);
+
+    enum PersistentEditorState {
+        PersistentEditor_NotOpen,
+        PersistentEditor_Opening,
+        PersistentEditor_Open,
+        PersistentEditor_ShouldRestore,
+        PersistentEditor_InDeferredRestore
+    };
```

Comment:
We try not to use bare enums in new codebase, but instead [scoped enum](https://www.geeksforgeeks.org/enum-classes-in-c-and-their-advantage-over-enum-datatype/)

```suggestion
    enum PersistentEditorState {
        NotOpen,
        Opening,
        Open,
        ShouldRestore,
        InDeferredRestore
    };
```

This will allow you to use `PersistentEditorState::Open` instead of `PersistentEditor_Open` in the implementation code

Reply:
Yeah, this code was written before I learned about the existence of `enum class`.